### ISSUE TITLE
DateTimeImmutable ignores timezones when useMicrosecondTimestamps() is set

### ISSUE THREAD

**Issue by dhm80** [Seldaek/monolog#832]
If I set a timezone to "non UTC" and enable using microseconds the timezone is completely ignored:

```php
$logger = new Logger('console');
$logger->setTimezone(new DateTimeZone('Europe/Berlin'));
$logger->useMicrosecondTimestamps(true);
$logger->pushHandler(new StreamHandler('php://stdout'));
$logger->debug("Test");
```

The problem it this part here I think:
https://github.com/Seldaek/monolog/blob/cc1fe9fa91de4a9c98a244b6f7d1302c6b0ad656/src/Monolog/DateTimeImmutable.php#L27-L33

Only if I set something like this as well the correct time would be logged:
```php
date_default_timezone_set("Europe/Berlin");
```

But that can't be the solution as I would overwrite global settings and may want the logger to have it's only timezone... that's what the parameter / method is for, right? ;)

**Comment by Seldaek** [comment]
I tried to fix this in #853 - which works if your default timezone is UTC, but otherwise it fails, at least for CET.. for Europe/Zurich for example it works, but I don't get why. Don't have time to dig further right now.

**Comment by Seldaek** [comment]
OK I found a way using an intermediate datetime object, it's not pretty but works.
